Sources/Hummingbird/Middleware/MiddlewareModule/_Middleware2.swift

```diff
@@ -0,0 +1,49 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A middleware that composes two other middlewares.
+///
+/// The two provided middlewares will be chained together, with `M0` first executing, followed by `M1`.
+///
+/// You won't typically construct this middleware directly, but instead will use result builder syntax.
+///
+/// ```swift
+/// router.addMiddleware {
+///    MiddlewareOne()
+///    MiddlewareTwo()
+/// }
+/// ```
+public struct _Middleware2<M0: MiddlewareProtocol, M1: MiddlewareProtocol>: MiddlewareProtocol where M0.Input == M1.Input, M0.Context == M1.Context, M0.Output == M1.Output {
```

Comment:
Can you add `@_documentation(visibility: internal)`. This isn't a symbol we need to publish documentation for.